### PR TITLE
Admin Profile Page 

### PR DESCRIPTION
**Pull request by kellyel**
## Proposed changes
This PR
- Adds an admin profile page (student version on the way)
- Allow users to change their own password
- Allow users to reset password via email
- Allow users to link / unlink SSO logins to their account
- Allow users to change their own user info
- Allow users to create / delete their password (delete only if another login method is available)

## Types of changes

What types of changes does this pull request introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (non-breaking change that does not add functionality but makes code cleaner or more efficient)
- [ ] Documentation Update
- [ ] Tests (new or updated tests)
- [ ] Style (changes to code styling)
- [ ] CI (continuous integration changes)
- [ ] Repository Maintenance
- [ ] Other (please describe below)

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/yeatmanlab/roar-dashboard/blob/main/.github/CONTRIBUTING.md).
- [x] The changes in this PR are as small as they can be. They represent one and only one fix or enhancement.
- [x] Linting checks pass with my changes.
- [ ] Any existing unit tests pass with my changes.
- [ ] Any existing end-to-end tests pass with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] If this PR fixes an existing issue, I have added a unit or end-to-end test that will detect if this issue reoccurs.
- [x] I have added JSDoc comments as appropriate.
- [x] I have added the necessary documentation to the [roar-docs repository](https://github.com/yeatmanlab/roar-docs).
- [x] I have shared this PR on the roar-pr-reviews channel (if I have access)
- [x] I have linked relevant issues (if any)

## Justification of missing checklist items
Need to add tests for changing userdata / password. Need to think about how to implement those tests.

## Further comments
This PR is still a Draft. TODO items are:
- [x] Overload the unlink function to allow users to unset their password
- [x] Overhaul the modal on the signin page, allowing the user to be more helpful to the user
- [x] Add support for students viewing their own data

### CODE REVIEW

File: src/components/NavBar.vue [hunk 1]
```diff
@@ -25,6 +25,12 @@
                 <div v-if="isWideScreen" class="nav-user-wrapper flex align-items-center gap-2 bg-gray-100">
                   <div class="text-lg font-bold text-gray-600">
                     {{ userDisplayName }}
+                    <router-link v-if="isAdmin" :to="{ name: 'Profile' }"
+                      ><button
+                        class="no-underline h-2 p-1 m-0 text-primary border-none border-round h-2rem text-sm hover:bg-red-900 hover:text-white"
+                      >
+                        <i class="pi pi-cog"></i></button
```
[the remark this answers is not in the document]
Comment (lucasxsong) [reply src/components/NavBar.vue:32]: I've implemented this Menu button [here](https://github.com/yeatmanlab/roar-dashboard/pull/707/files#diff-c3ceb15a511b3e08b33b8862b4023b07895ad41b7fbbceffc8802d5ae0e964e8R32), but in short, the component itself is 

```
<PvButton
                    type="button"
                    icon="pi pi-cog"
                    @click="toggleSettingsMenu"
                    aria-haspopup="true"
                    aria-controls="overlay_menu"
                    class="m-0 text-primary border-none border-round h-2rem text-sm hover:bg-red-900 hover:text-white"
                  />
                  <PvMenu ref="settingsMenu" id="settings-menu" :model="settingItems" :popup="true" />
```

And the script section is 


```
const settingsMenu = ref();
const toggleSettingsMenu = (event) => {
  settingsMenu.value.toggle(event);
};
const settingItems = [
  {
    label: 'Update Settings',
    icon: 'pi pi-cog',
    command: () => {
      router.push({ name: 'UserSettings' });
    },
  },
  {
    label: 'Sign Out',
    icon: 'pi pi-sign-out',
    command: () => {
      router.push({ name: 'SignOut' });
    },
  },
];
```

Note that the route may need to be changed, as I had used a different route name. Happy to riff on the route name as well.

**Review by lucasxsong — CHANGES_REQUESTED** [review]

**Comment by lucasxsong** [comment]
Thank you for putting this up! A manage profile component has been long needed and you have set a great foundation. I've left some comments, mostly minor. Let me know if you have anything you want to talk about.

**Review by lucasxsong — APPROVED** [review]
This is looking great!

**Review by richford — APPROVED** [review]

**Comment by richford** [comment]
LGTM! I noticed that your original comment you said you still have to do

> - [ ] Add support for students viewing their own data

Should we hold off on that or merge now?

**Comment by kellyel** [comment]
I'm going to merge this, as there's some other changes that need to be made to allow students into the profile page, namely locking down the rest of the profile page from them (change password, providers). My initial attempt at this resulted in a bit of inconsistency in the way I was grabbing userType, so I'm going to do a small follow up PR.

**Comment by kellyel** [comment]
Upon further review, I think I've solved that issue. That item is now included in this PR.

**Pull request merged**